### PR TITLE
[Fixed sessions] Reveal cluster information on click and retain default cluster zoom in behavior

### PR DESCRIPTION
**Pull request by Sonje**
- I used built in option to intercept cluster selection from the google maps library.
- We are calculating exact cluster position through `getClusterPosition`utility function.
- I decided to use SVG component for zoom in icon for easy and performative switch between colors (no need to load different assets)
- I created a cluster slice to handle API request and track cluster visibility state (we are using opacity transition for smoother visual outlook when component changes visibility)
- The design for mobile was way different from dekstop so I decided to go with the different elements to properly render styles based on the device.
- Decided to not include tablet specific design
- I added some click/touchend/grab handling to make sure the cluster disappears when it should. 
- When a user clicks on zoom icon we are using the default cluster click behaviour from maps library.

Please let me know if anything's unclear. 
Screen recordings of the functionality attached (old masters map refresh glitch):

https://github.com/user-attachments/assets/46da77fe-995e-49b2-bbde-9cfa1fc05a8b

https://github.com/user-attachments/assets/b976f23d-a58b-4ec9-9fc0-82d00bad92a7

### CODE REVIEW

**Comment by Sonje** [comment]
[fixed: clicking cluster](https://trello.com/c/XasyvVfG/1816-fixed-clicking-cluster)

**Comment by Sonje** [comment]
I accidentally merged in main branch instead of rebasing, but since we are doing squash and merge that should not change much, let me know if that's a problem, I can always hard reset and rebase if needed.

File: app/javascript/react/components/Map/Markers/ClusterInfo/ClusterZoomIn/ClusterZoomIn.style.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+const ZoomInContainer = styled.div<ZoomInContainerProps>`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: ${(props) => (props.$isMobile ? "2.8rem" : "4.2rem")};
+  height: 2.8rem;
+  border-radius: 25px;
+  background-color: ${white};
+  cursor: pointer;
+  ${(props) => !props.$isMobile && "box - shadow: 2px 2px 4px 0px #4c56601a;"}
```
Comment (evulins) [inline app/javascript/react/components/Map/Markers/ClusterInfo/ClusterZoomIn/ClusterZoomIn.style.tsx:17]: I think we have gray900 for this hex color :D

Comment (Sonje) [reply app/javascript/react/components/Map/Markers/ClusterInfo/ClusterZoomIn/ClusterZoomIn.style.tsx:17]: will change it promptly

**Pull request merged**